bigquery/google/cloud/bigquery/client.py

```diff
@@ -1202,6 +1209,13 @@ def query(
         Keyword Arguments:
             job_config (google.cloud.bigquery.job.QueryJobConfig):
                 (Optional) Extra configuration options for the job.
+            merge_job_config (bool):
```

Comment:
I hadn't thought to add an option to turn off the default overrides. We can disambiguate between a value in the `job_config` which is unset (no key in the dictionary) versus a value which is explicitly set to `None` (key in dictionary, with value of `None`), so this may not be necessary. Folks can override defaults by setting the value in the specified config.

What was the use case you are proposing for this argument?

Reply:
Yes that was exactly the thinking. I wasn't sure if None was a valid value for some of these keys, but now as I look through at many of the `setters` it seems it won't get made (and you would know better than I would ha). I'll try it and see if the tests pass.

Reply:
It works, but I'm a little torn about the convenience of having to manually set everything to `None` in order to override. It definitely leads to a cleaner function signature, but if they have a default with a lot of values then that might get a little tedious. I'll leave it up to you guys which version we go with.

Reply:
I'd prefer to leave it out for now. If someone creates a client with a whole bunch of default properties, I think it's natural that they'd have to override them. (I know you are thinking of use cases where the person running the queries isn't necessarily the person that constructs the client, but hopefully you aren't setting too many default properties in that case.)

Reply:
:+1: